File: entites/SideDef.py
class SideDef:
    def __init__(self, x, y, upperTextureName, lowerTextureName, middleTextureName, sectorNumber):
        self.x = x
        self.y = y
        self.upperTextureName = upperTextureName
        self.lowerTextureName = lowerTextureName
        self.middleTextureName = middleTextureName
        self.sectorNumber = sectorNumber

    def __str__(self):
        return f"{self.x = }, {self.y = }, {self.upperTextureName = }, {self.lowerTextureName = }, {self.middleTextureName = }, {self.sectorNumber = }"

    __repr__ = __str__

File: cpp_code/cpp_classes/SideDef.cpp
#include "../headers/SideDef.h"

SideDef::SideDef(uint16_t x, uint16_t y, uint16_t sectorNumber, std::string upperTextureName, std::string lowerTextureName, std::string middleTextureName)
        : x(x), y(y), sectorNumber(sectorNumber), upperTextureName(upperTextureName), lowerTextureName(lowerTextureName), middleTextureName(middleTextureName)
    {}
std::ostream& operator<<(std::ostream& os, const SideDef& obj) {
    os << "SideDef{ ";
    os << "x: " << (int) obj.x << " ";
    os << "y: " << (int) obj.y << " ";
    os << "upperTextureName: " << obj.upperTextureName << " ";
    os << "lowerTextureName: " << obj.lowerTextureName << " ";
    os << "middleTextureName: " << obj.middleTextureName << " ";
    os << "sectorNumber: " << (int) obj.sectorNumber << " ";
    os << "}";
    return os;
}